plugins/inventory/proxmox.py

```diff
@@ -275,32 +275,33 @@ def _get_session(self):
         return self.session
 
     def _get_auth(self):
-
         validate_certs = self.get_option('validate_certs')
 
         if validate_certs is False:
             from requests.packages.urllib3 import disable_warnings
             disable_warnings()
 
         if self.proxmox_password:
-
             credentials = urlencode({'username': self.proxmox_user, 'password': self.proxmox_password})
-
             a = self._get_session()
-
             ret = a.post('%s/api2/json/access/ticket' % self.proxmox_url, data=credentials)
-
             json = ret.json()
-
             self.headers = {
                 # only required for POST/PUT/DELETE methods, which we are not using currently
                 # 'CSRFPreventionToken': json['data']['CSRFPreventionToken'],
                 'Cookie': 'PVEAuthCookie={0}'.format(json['data']['ticket'])
             }
-
         else:
+            # Clean and format token components
+            user = self.proxmox_user.strip()
+            token_id = self.proxmox_token_id.strip()
+            token_secret = self.proxmox_token_secret.strip()
```

Comment:
I'm wondering whether this is a good idea. Is `token_secret` a user-selected string that can potentially have valid whitespace in it, that would get incorrectly removed by this change? (Most likely spaces, and not newlines.)

Reply:
Normally, pve tokens should never have spaces. If this is the case the token will be invalid in all cases.
https://pve.proxmox.com/wiki/Proxmox_VE_API#Example:_Use_API_Token
Exemple with token format
```
curl -H "Authorization: PVEAPIToken=root@pam!monitoring=aaaaaaaaa-bbb-cccc-dddd-ef0123456789" https://10.0.0.1:8006/api2/json/
```

Reply:
Hmm, what about token IDs and usernames? Can they have spaces?

Reply:
i try to create local user with space. It's not possible. 
```
userid: invalid format - value 'user name@pam' does not look like a valid user name
```
(i cant try with other type (LDAP/AD/OpenID))


we can remove strip to self.proxmox_user if you prefer.

Reply:
Did you also test token ID? If that doesn't work either, everything should be fine.

Reply:
I try for token ID space is not possible : 
```
Method 'POST /access/users/ansible@pam/token/user space' not implemented (501
```